### DISCUSSION TITLE
Thoughts on single-line admonitions?

### DISCUSSION THREAD

**Discussion by JoshuaKGoldberg** [facebook/docusaurus#8458] [General] [answered]
Porting from https://github.com/typescript-eslint/typescript-eslint/pull/6251#discussion_r1051663765 with @Josh-Cena - although a `:::note` or `:::tip` may sometimes be the _semantically_ correct approach for subtly emphasized content, _visually_ they take up a lot of space.

<img width="493" alt="Screenshot of a tip admonition and a quote both saying 'Subtly emphasized content.', interspersed with regular paragraphs saying 'Regular text.'" src="https://user-images.githubusercontent.com/3335181/208319909-f741c90d-3a4b-4ceb-be3f-e3c019a062aa.png">

Really, I don't want to include the _ℹ️ NOTE_ for a small note. Does this feel reasonable as a feature request?

<img width="489" alt="Mockup of a note admonition without its indicator" src="https://user-images.githubusercontent.com/3335181/208320191-ae835ebe-5a4f-40e3-b724-4a5679efceb3.png">

A couple starting thoughts on how Docusaurus could add this feature:

* _(My top choice)_ Allowing passing props to admonitions like `:::note single-line`, the same way you can provide a filename to a codeblock like <code>```js title="index.js"</code>
* Providing a new admonition type, like `:::single-line`

**Comment by slorber** [comment, marked as answer]
With MDX 2 we are going to use Markdown Directives (https://talk.commonmark.org/t/generic-directives-plugins-syntax/444)

It's not std but is more likely to become std than any other fancy md syntax that we can invent, and there's an official remark plugin to parse that: https://github.com/remarkjs/remark-directive

I also thought single-line admonitions make sense , and we can implement that after the MDX 2 upgrade

```
:note[Note text]
```

---

Note: remark directives work in 2 steps:
- remark-directives parse the directives as AST nodes (generic/shared)
- one/many other remark plugin transform those directive AST nodes to something else (unprocessed AST directives are printed back as is in the original doc)

That makes it easier to implement your own custom directives

**Reply by JoshuaKGoldberg** [reply]
Fantastic, thanks!
